src/jdk.incubator.jextract/share/classes/jdk/incubator/jextract/tool/resources/C-X.java.template

```diff
@@ -25,7 +25,7 @@ public class C-X {
 
     private static void copy(MemoryAddress addr, ${CARRIER}[] arr) {
         var heapSegment = MemorySegment.ofArray(arr);
-        MemoryAddress.copy(heapSegment.baseAddress(), addr, arr.length * sizeof());
+        addr.segment().copyFrom(heapSegment);
```

Comment:
The behavior is not necessarily the same if `addr` is not the base address of the segment. I think the right move here is to change the parameter from MemoryAddress to MemorySegment (same for Cstring).